specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,278 @@
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Non-standard operation."
+@summary("Get specific schema versions.")
+@doc("Gets one specific version of one schema.")
+@route("/$schemaGroups/{groupName}/schemas/{schemaName}/versions/{schemaVersion}")
+@get
+op getSchemaByVersion is Azure.Core.Foundations.Operation<
+  {
+    @doc("Name of schema group.")
+    @path
+    groupName: string;
+
+    @doc("Name of schema.")
+    @path
+    schemaName: string;
+
+    @doc("Version number of specific schema.")
+    @path
+    schemaVersion: int32;
+  },
+  OkResponse & Schema
+>;
+
+//  POST https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{schemaName}:get-id?api-version={apiVersion}
+// Operation name more accurately reflects the return value than the action
+#suppress "@azure-tools/typespec-azure-core/byos" "storage service is not used"
+@summary("Get properties for existing schema.")
+@doc("Gets the properties referencing an existing schema within the specified schema group, as matched by schema content comparison.")
+@action("get-id")
+op getSchemaPropertiesByContent is resourceOperations.ResourceAction<
+  SchemasName,
+  {
+    ...SchemaContentType;
+    ...SchemaContent;
+  },
+  SchemaProperties
+>;
+
+//  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{schemaName}?api-version={apiVersion}
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Non-standard operation."
+#suppress "@azure-tools/typespec-azure-core/byos" "storage service is not used"
+@summary("Register new schema")
+@doc("Register new schema. If schema of specified name does not exist in specified group, schema is created at version 1. If schema of specified name exists already in specified group, schema is created at latest version + 1.")
+@route("/$schemaGroups/{groupName}/schemas/{schemaName}")
+@put
+op registerSchema is Azure.Core.Foundations.Operation<
+  {
+    @doc("Name of schema group.")
+    @path
+    groupName: string;
+
+    @doc("Name of schema.")
+    @path
+    schemaName: string;
+
+    ...SchemaContent;
+    ...SchemaContentType;
+  },
+  NoContentResponse<SchemaProperties>
+>;
+
+// Models
+alias NoContentResponse<T> = TypeSpec.Http.Response<204> & T;
+
+union SchemaContentTypeValues {
+  avro: "application/json; serialization=Avro",
+
+  @added(ServiceVersion.V2022_10)
+  json: "application/json; serialization=Json",
+
+  @added(ServiceVersion.V2022_10)
+  custom: "text/plain; charset=utf-8",
+
+  @added(ServiceVersion.V2023_07_01)
+  protobuf: "text/vnd.ms.protobuf",
+}
+
+alias SchemaContentType = {
+  @doc("The content type for given schema.")
+  @header("Content-Type")
+  contentType: SchemaContentTypeValues;
+};
+
+alias SchemaContent = {
+  @body
+  @doc("String representation (UTF-8) of the schema.")
+  content: bytes;
```

Comment:
This should be `schemaContent` per the original swagger: 
- https://github.com/Azure/azure-rest-api-specs/blob/c577d8f27e0c3e5e4885134db19bc8193af7260c/specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2023-07-01/schemaregistry.json#L426
- https://github.com/Azure/azure-rest-api-specs/blob/c577d8f27e0c3e5e4885134db19bc8193af7260c/specification/schemaregistry/data-plane/Microsoft.EventHub/stable/2023-07-01/schemaregistry.json#L337

Reply:
Per offline discussion, this is not a necessary change.